internal/pkg/service/stream/dependencies/source.go

```diff
@@ -91,9 +91,9 @@ func newSourceScope(svcScp ServiceScope, distScp dependencies.DistributionScope,
 		return nil, err
 	}
 
-	d.Plugins().RegisterSinkPipelineOpener(func(ctx context.Context, sinkKey key.SinkKey, sinkType definition.SinkType) (pipeline.Pipeline, error) {
+	d.Plugins().RegisterSinkPipelineOpener(func(ctx context.Context, sinkKey key.SinkKey, sinkType definition.SinkType, onClose func()) (pipeline.Pipeline, error) {
 		if d.Plugins().IsSinkWithLocalStorage(sinkType) {
-			return d.storageRouter.OpenPipeline(ctx, sinkKey)
+			return d.storageRouter.OpenPipeline(ctx, sinkKey, onClose)
 		}
```

Comment:
Close of a pipeline can be now triggered by the disk writer node shutdown, so it is now async,... solved by the `onClose` callbacks.

Reply:
The `onClose` callback notifies the parent layer about the change.